src/Processors/Transforms/buildPushingToViewsChain.cpp

```diff
@@ -299,8 +306,19 @@ Chain buildPushingToViewsChain(
 
         if (auto * materialized_view = dynamic_cast<StorageMaterializedView *>(dependent_table.get()))
         {
+            auto lock = materialized_view->tryLockForShare(context->getInitialQueryId(), context->getSettingsRef().lock_acquire_timeout);
+
+            if (lock == nullptr)
+            {
+                LOG_WARNING(
+                    &Poco::Logger::get("PushingToViews"),
+                    "Trying to access table {} but it doesn't exist",
```

Comment:
You have shared_ptr to the table, why it doesn't exist?

Reply:
Because it has been deleted between the acquisition and the use of the shared_ptr by another thread (the one dropping the Materialized View)

Reply:
Why do you think that is has been deleted? Maybe we just failed to acquire the lock due to `lock_acquire_timeout`

Reply:
It is currently giving two kind of exceptions (reproduced by the new test): UNKNOWN_TABLE and TABLE_IS_DROPPED, so I assumed that there was no timeout.
Anyway, I can modify the WARNING message including that possibility